### PR TITLE
Add numpydoc

### PR DESCRIPTION
**Pull request by jschueller**
Fixes https://github.com/conda-forge/staged-recipes/issues/2165

### CODE REVIEW

File: recipes/numpydoc/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    - python
+    - sphinx
+
+test:
+  imports:
+    - numpydoc
+
+about:
+  home: https://github.com/numpy/numpydoc
+  license: BSD
```
Comment (jakirkham) [inline recipes/numpydoc/meta.yaml]: Please specify the number of clauses (e.g. BSD 3-Clause).

Comment (jakirkham) [inline recipes/numpydoc/meta.yaml]: Optionally can add `license_family: BSD`.

File: recipes/numpydoc/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+source:
+  fn: numpydoc-{{ version }}.tar.gz
+  url: https://github.com/numpy/numpydoc/archive/v{{ version }}.tar.gz
+  sha256: 7a14bd403f1fd96d3e27af2b0679312b7931698a3c112bfc240bc032ae6a0e4a
+
+build:
+  number: 0
+  script:
+    - python setup.py install --single-version-externally-managed --record=record.txt
```
Comment (jakirkham) [inline recipes/numpydoc/meta.yaml]: Could inline this.

**Comment by jakirkham** [comment]
Looks pretty good, @jschueller . Thanks for adding this. Couple easy suggestions above.

File: recipes/numpydoc/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+about:
+  home: https://github.com/numpy/numpydoc
+  license: BSD
+  license_file: LICENSE.txt
+  summary: Numpy's Sphinx extensions
+
+extra:
+  recipe-maintainers:
+    - jschueller
```
Comment (scopatz) [inline recipes/numpydoc/meta.yaml]: Also feel free to add me as a maintainer

Comment (jakirkham) [inline recipes/numpydoc/meta.yaml]: Done below.

**Comment by jakirkham** [comment]
cc @rgommers

File: recipes/numpydoc/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,38 @@
+{% set version = "0.6" %}
+
+package:
+  name: numpydoc
+  version: {{ version }}
+
+source:
+  fn: numpydoc-{{ version }}.tar.gz
+  url: https://github.com/numpy/numpydoc/archive/v{{ version }}.tar.gz
```
Comment (rgommers) [inline recipes/numpydoc/meta.yaml]: This is generally not the right thing to do. This link points to a tarball automatically created by GitHub, which is not the same as what one gets from ``python setup.py sdist`` and uploads to PyPI. For ``numpy`` and ``scipy`` we tend to fix that by uploading tarballs to GitHub Releases, but not for ``numpydoc``. So link should be to PyPI.

**Comment by rgommers** [comment]
LGTM aside from the link. Thanks for making this PR.

**Comment by jakirkham** [comment]
Thanks everyone for taking a look and @jschueller for putting it together.

**Pull request merged**